koap/build.gradle.kts

```diff
@@ -26,7 +27,16 @@ tasks.withType<JacocoReport> {
 
 kotlin {
     jvm()
-    js().browser()
+    js {
+        apply(from = "${rootDir}/gradle/npm.gradle")
+        browser {
+            distribution {
+                file("${buildDir}/npmdist/")
+            }
+        }
+        binaries.executable()
+        // target = "v5"
```

Comment:
I haven't been able to figure out the syntax to specify the target java version here...

Reply:
> target java version

I assume you meant **javascript**.

We can specify the target Java version via (this can go at the top-level of this file, i.e. does not need to be under `kotlin { jvm { ... } }`):

```kotlin
tasks.withType<KotlinCompile>().all {
    kotlinOptions {
        jvmTarget = "1.8"
    }
}
```

Maybe worth trying a variant of the above snippet and seeing if there is a similar Javascript version?

Reply:
Sorry of the confusion, I mis-typed I meant javascript not java ☕ 

I was trying to specify the ECMA target as specified here:
https://kotlinlang.org/docs/reference/using-gradle.html#attributes-specific-for-js

Reply:
We can pass compiler arguments to the Kotlin compiler. So we'll probably have to figure out what the previous `target = "v5"` was resolving to as far as a compiler arg.